### DISCUSSION TITLE
Batch-downloading APSOIL or SLGA datasets using Phyton or R

### DISCUSSION THREAD

**Discussion by KeLiu7** [APSIMInitiative/ApsimX#8833] [General]
APSIM Next Gen GUI provides an interface for downloading soil propertities from APSOIL, SLGA and ISRIC, but it seems that only ISRIC has the R package that can be used to quickly replace soil. Just wondering if there is a convenient way to download the APSOIL and SLGA databases as well? @par456 @femiguez

**Comment by femiguez** [comment]
@KeLiu7 You can download soils with the APSoil app. They will be stored as .soils (XML) files. You can load them into R using the 'read_apsim_soils' function. If you are trying to build simulations over a grid there is also soil data in https://worldmodel.csiro.au/. You can (for example) get these soils using the 'apsimx:::get_worldmodeler_soil_profile' function.

**Reply by KeLiu7** [reply]
Thanks Fernando - Yes, I'm trying to build simulations over grid within Australia. https://worldmodel.csiro.au/ this website does not work.
APSoil app works but too slow and could only be used on iPad. Do you have any other ideas?

**Reply by femiguez** [reply]
worldmodel works just fine. You can create a grid using the 'get_worldmodeler_soil_profile' function in R. I haven't used APSoil lately, but I'm assuming it still works for Windows

**Reply by KeLiu7** [reply]
Yes, indeed no problem with the global soil database, but my research scope is concentrated in Australia, so APSoil database might be more suitable, but if APSoil is not suitable for grid simulations, I will use get_worldmodeler_soil_profile or get_isric_soil_profile. Could you please confirm this -  if APSoil is not suitable for grid simulations?

**Reply by femiguez** [reply]
These are synthetic soils anyway, so I can't say that APSoil is not appropriate for grid simulations. It appears that worldmodeler is more reliable than isric. I do not know if either one generates 'better' soil profiles, it probably depends on your specific needs
